hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneFileSystem.java

```diff
@@ -783,5 +797,11 @@ private FileStatus convertFileStatus(
         fileStatusAdapter.getPath()
     );
 
+    BlockLocation[] blockLocations = fileStatusAdapter.getBlockLocations();
+    if (blockLocations == null || blockLocations.length == 0) {
```

Comment:
Why we need this special handling. Even in this case, we can return LocatedFileStatus with an empty right, like using default BlockLocation constructor. Or is there any reason for doing this way.

Reply:
When we send LocatedFileStatus without the locations, I felt that it seemed wrong. If we cannot get block locations, we go to default behavior of base filesystem class.

Reply:
QQ: If block location is empty it is 0-byte key. So, sending as before will help anyway?